Update README

uncomputable opened the pull request:
Structure, update and compress the building instructions for the C, Haskell and Coq projects.

roconnor-blockstream commented:
This is great.  Are you willing to add a link to rust-simplicity as well somewhere?  Maybe in the Further Resources section?

uncomputable commented:
Good idea; I added a reference.

The slides from the BPASE 2018 presentation are no longer available, so I removed the dead link. Feel free to upload the slides somewhere if you still have them.
